mixins/form-validation.js

```diff
@@ -0,0 +1,109 @@
+import { getAllValues } from '@/utils/object';
+import formRulesGenerator from '@/utils/validators/formRules/index';
+
+export default {
+  data() {
+    return {
+      formRulesets: [
+        /* formRulesets define the validation rules for the entire form. These should almost always be overridden in the form-component using the mixin
+        example:
+        {
+          path:  'spec.template.spec.containers.securityContext.runAsUser', // required: defines the path of the value to be tested against it's rules. Uses this.value unless overridden via optional rootObject
+          rootObject: { container: { image: 'name' } } // optional: redirects the path to the object passed here,
+          rules: ['noSpaces', 'noPeriods'], required: array of functions to run against the value of the field defined by the path (and optionally the rulesets rootObject),
+          translationKey: '', // optional: defines the displaykey, overrides displaykeys that may otherwise be passed into the translation
+        }
+        */
+      ],
+      boundValidationPaths: [
+        // an array of strings that track which ruleset paths have been bound to a field for error reporting
+        // tracked in a seperate array from the actual rulesets since we want the option of keeping track of modelValidationRules without mutating the model itself
+        // you may place a path in here manually as part of your form's data props...
+        // ... or you may place a path in here programitcally by using the "getAndBindPathRules" method
+      ]
+    };
+  },
+
+  methods: {
+    getPathRules(path) {
+      return this.rulesets.find(ruleset => ruleset.path === path)?.rules || [];
+    },
+    getAndBindPathRules(path) {
+      const rules = this.getPathRules(path);
+
+      if (rules.length > 0 && !this.boundValidationPaths.includes(path)) {
+        this.boundValidationPaths = [...this.boundValidationPaths, path];
+      }
+
+      return rules;
+    },
+    getPathValues(path) {
+      const relevantRuleset = this.rulesets.find(ruleset => ruleset.path === path);
```

Comment:
why does this get the rules set for the path by filtering rulesets instead of by using `this.getPathRules(path)`?

Reply:
getPathRules only returns the rules on the ruleset whereas I need the whole ruleset here since I need to check for the optional `rootObject` property.